src/serde/room_state.rs

```diff
@@ -0,0 +1,104 @@
+use std::fmt;
+
+use serde::{
+    de::{self, Deserializer, SeqAccess, Visitor},
+    ser::{SerializeSeq, Serializer},
+};
+use serde_json::Value as JsonValue;
+
+use crate::v1::join::create_join_event::RoomState;
+
+pub fn serialize<S>(room_state: &RoomState, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut seq = serializer.serialize_seq(Some(2))?;
+    seq.serialize_element(&200)?;
+    seq.serialize_element(room_state)?;
+    seq.end()
+}
+pub fn deserialize<'de, D>(deserializer: D) -> Result<RoomState, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    deserializer.deserialize_seq(RoomStateVisitor)
+}
+
+struct RoomStateVisitor;
+
+impl<'de> Visitor<'de> for RoomStateVisitor {
+    type Value = RoomState;
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("Room State response")
+    }
+
+    fn visit_seq<V>(self, mut seq: V) -> Result<Self::Value, V::Error>
+    where
+        V: SeqAccess<'de>,
+    {
+        let bogus = seq
+            .next_element::<u32>()?
+            .ok_or_else(|| de::Error::custom("Expected integer as first element"))?;
+        if bogus != 200 {
+            panic!("Expected 200");
+            // return Err(SeqAccess::Error::Formatter);
+        }
+
+        let room_state_map = seq
+            .next_element::<JsonValue>()?
+            .ok_or_else(|| de::Error::invalid_length(1, &self))?;
+        Ok(serde_json::from_value::<RoomState>(room_state_map).unwrap())
```

Comment:
There should be no need to go through `JsonValue` here.

```suggestion
        seq.next_element()
```